nBTC/sources/verify_payment.move

```diff
@@ -38,7 +38,8 @@ public fun verify_payment(
     outputs.do!(|o| {
         if (o.script_pubkey() == receiver_script_pubkey) {
             amount = amount + o.amount();
-        } else if (o.is_op_return()) {
+        } else if (o.is_op_return() && op_msg.is_none()) {
+            // we select the first OP RETURN
             op_msg = o.op_return();
         }
     });
```

Comment:
probably we should change to for loop, and have early break once we find op_return.